web-registry/src/hooks/bridge/useBridged.ts

```diff
@@ -1,36 +1,51 @@
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
 
-import { TablePaginationParams } from 'web-components/lib/components/table/ActionsTable';
-
-import { BatchInfoWithBalance } from 'types/ledger/ecocredit';
-
-// TODO - this hook is a placeholder that right now just simulates a request to
-// the network and an empty response. Use later to implement the data request.
-
-async function stall(stallTime = 3000): Promise<void> {
-  await new Promise(resolve => setTimeout(resolve, stallTime));
-}
+import { useAllCreditClassQuery } from 'generated/sanity-graphql';
+import { BridgedEcocredits } from 'types/ledger/ecocredit';
+import { getBridgedEcocreditsForAccount } from 'lib/ecocredit/api';
+import { client as sanityClient } from 'sanity';
 
 interface Props {
   address?: string;
-  paginationParams?: TablePaginationParams;
 }
 
 interface Output {
-  credits: BatchInfoWithBalance[];
+  bridgedCredits: BridgedEcocredits[];
   isLoadingCredits: boolean;
 }
 
-export const useBridged = ({ address, paginationParams }: Props): Output => {
-  const [credits, setCredits] = useState<BatchInfoWithBalance[]>([]);
+export const useBridged = ({ address }: Props): Output => {
+  const [bridgedCredits, setCredits] = useState<BridgedEcocredits[]>([]);
   const [isLoadingCredits, setIsLoadingCredits] = useState(true);
+  const isFetchingRef = useRef(false);
```

Comment:
I know we use this pattern in the app but it was more like a bandaid to prevent overfetching when didn't have react-query.
Using react-query allows for dealing with this issue while at the same time populating the cache and simplifying the code.
For those reasons, it would be best to create new queries with react-query. I think it can be a separate issue since dev is already done here but it would reduce our workload to use react-query by default in the future.

Reply:
I guess we would need to set up different pages since right now we have the 2 tabs (bridgable and bridged ecocredits tables) on the same /bridge page?

Reply:
Oh ok, we could use react-query even with one page (without a loader) but no problem to have that in a separate PR. Was just a thought for future tasks.

Reply:
right, I had the react-query/react-router issues in mind ;)
but will definitely at least use react-query in a follow-up PR, we could do that as part of the follow-up issue for adding more info to the table: https://github.com/regen-network/regen-web/pull/1586#issuecomment-1325572481

Reply:
right, I had the react-query/react-router issues in mind ;)
but will definitely at least use react-query in a follow-up PR, we could do that as part of the follow-up issue for adding more info to the table: https://github.com/regen-network/regen-web/pull/1586#issuecomment-1325572481